### ISSUE TITLE
buildState is slow on big deployments

### ISSUE THREAD

**Issue by seb4stien** [Praqma/helmsman#418]
The buildState is quite slow on big deployments (> 300 apps) due to the getReleaseContext on each release. It can takes a few minutes which has a big impact on CI/CD pipelines (because we have to build the state even for simple changes).

A possible solution could be to retrieve all the contexts per namespace using a query like: 
```
kubectl get secret -l owner=helm -o jsonpath='{range .items[*]}{.metadata.name}"-"{.metadata.labels.HELMSMAN_CONTEXT}"|"{end}'
```

Do you already have some plan to work on this part? What do you think of this proposal?

**Comment by wadi-chemkhi** [comment]
I'm having the same issue... Did you find a temporary workaround @seb4stien ?

**Comment by sami-alajrami** [comment]
Just for the record, this has been addressed with PR #420
